libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -1605,6 +1605,15 @@ void test_MQTT_ProcessLoop_handleIncomingAck_Error_Paths( void )
                             MQTTIllegalState, MQTTBadResponse, MQTTStateNull,
                             MQTTBadResponse, false, NULL );
 
+    /* Verify that MQTTServerRefused is propagated when SUBACK indicates a
+     * server refusal. */
+    currentPacketType = MQTT_PACKET_TYPE_SUBACK;
+    isEventCallbackInvoked = false;
+    expectProcessLoopCalls( &context, MQTTServerRefused, MQTTStateNull,
+                            MQTTIllegalState, MQTTBadResponse, MQTTStateNull,
+                            MQTTServerRefused, false, NULL );
+    TEST_ASSERT_TRUE( isEventCallbackInvoked );
```

Comment:
Can you also add the same test for the `UNSUBACK` packet as the new behavior will invoke the callback for server rejections in both subscribe and unscubscribe requests?

Reply:
No, since deserializing UNSUBACK can't return `MQTTServerRefused`, only `MQTTBadResponse` (or `MQTTBadParameter` if we called it improperly)

Reply:
So, can you add a test for invoking the callback with `MQTTBadResponse`  for UNSUBACK?